project_task_code/models/project_task.py

```diff
@@ -28,3 +29,41 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        tasks = []
+        code = self._name_search_parse_code(name)
+        if code:
+            # We are searching for codes
+            tasks = self.search(
+                [('code', '=ilike', code + '%')] + args,
+                limit=limit).name_get()
+            args += [('id', 'not in', [i[0] for i in tasks])]
```

Comment:
never change a mutable object you got passed as function parameter. I think this is also a weird place to do this. I'd assign this to a new variable before the last `return`